crates/matrix-sdk/src/client/mod.rs

```diff
@@ -2448,12 +2448,13 @@ impl Client {
         C: Future<Output = LoopCtrl>,
     {
         self.sync_with_result_callback(sync_settings, |result| async {
-            if let Ok(sync_response) = result {
-                callback(sync_response).await
-            } else {
-                warn!("Error from sync with result: {}", result.err().unwrap());
```

Comment:
Ouch, how did that end up there. I wonder if we could enable the unwrap clippy warning somehow. I guess not since we use it for locks and in tests.